Automattic/newspack-plugin
# Feat/deprecate test mode

### All Submissions:






### Changes proposed in this Pull Request:

Along with https://github.com/Automattic/newspack-popups/pull/365 this simplifies campaign settings in a few ways:

- Removes Test Mode. Recently added Preview and "preview as segment" functionality provides a richer approach and should replace Test Mode.
- Removes the "never" frequency. This setting is redundant as there are now better ways to keep campaign from being seen, including unpublishing and assigning to an inactive campaign group (or no campaign group).
- Adds "Unpublish" UI to the campaigns wizard, as a simpler alternative to the "never" frequency.

Dependent PRs:
https://github.com/Automattic/newspack-plugin/pull/775
https://github.com/Automattic/newspack-plugin/pull/771
https://github.com/Automattic/newspack-plugin/pull/769
https://github.com/Automattic/newspack-plugin/pull/768
https://github.com/Automattic/newspack-plugin/pull/767

**Note: this is a breaking change because campaigns with frequencies of "never" or test mode will no longer work. This shouldn't be deployed until a migration strategy has been implemented.**

### How to test the changes in this Pull Request:

Switch `newspack-popups` to `feat/deprecate-test-mode`. 

1. Observe Test mode UI and "Never" frequency have been removed from the editor sidebar and the Newspack Campaigns wizard pop-overs.
2. Observe a new "Unpublish" options in the Campaigns Wizard pop-over, for campaigns that are published.
3. Verify clicking Unpublish moves the campaign to the "Drafts" grouping, and causes the campaign to not appear in group previews or the front end.
4. Verify individual campaigns can still be previewed, whether published or not.
5. Verify group previews continue to work as expected.
6. Verify front end display of campaigns continues to work as expected.
7. Switch to `master` braches (in both repositories) and set several campaigns to use Test mode and/or Never frequency. 
8. Switch back to these branches, verify these campaigns do not display in previews or the front-end.

### Other information:
